utils/include/tlv.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * @brief Write data to storage.
+ * 
+ * @param ctx Storage context.
+ * @param offset Location where to place the date.
+ * @param data Data to save.
+ * @param data_size Number of bytes to save.
+ * 
+ */
+typedef int (*tlv_storage_write)(void *ctx, uint32_t offset, uint8_t *data, uint32_t data_size);
+
+/**
+ * @brief Read data from storage.
+ * 
+ * @param ctx Storage context.
+ * @param offset Location where to find the date.
+ * @param data Buffor where data will be placed.
+ * @param data_size Number of bytes to read.
+ * 
+ */
+typedef int (*tlv_storage_read)(void *ctx, uint32_t offset, uint8_t *data, uint32_t data_cap);
+
+typedef struct tlv_ctx {
+	struct storage_impl {
+		void *ctx; // Context of the storage API
+		tlv_storage_write write; // Function to write data to storage
+		tlv_storage_read read; // Function to read data from storage
+	} storage; // API to access storage location.
+	uint32_t first_entry_offset; // First valid address where data can be placed.
+	uint32_t next_free_offset; // Address where next new element will be saved.
+	uint32_t last_valid_offset; // Address that point to the end of valid memory range.
+} tlv_ctx;
+
+typedef uint16_t tlv_type;
+typedef struct {
+	uint8_t padding;
+	uint8_t size;
+} tlv_size;
+#define DATA_ALIGN 4
+#define CALCULATE_PADDING(val) (DATA_ALIGN - (val % DATA_ALIGN))
+#define PADDING_BYTE 0xFF
+
+typedef struct {
+	tlv_type type;
+	tlv_size size;
+} tlv_header;
+
+bool tlv_lookup(tlv_ctx *ctx, tlv_type type, tlv_header *header);
```

Comment:
minor: in most ncs modules `int` value with errno is returned (0 in case success, negative error code otherwise) in such case

Reply:
ok, used errno for the API